src/azure-cli-core/azure/cli/core/mock.py

```diff
@@ -32,17 +32,23 @@ def __init__(self, commands_loader_cls=None, random_config_dir=False, **kwargs):
             self.env_patch = patch.dict(os.environ, {'AZURE_CONFIG_DIR': config_dir})
             self.env_patch.start()
 
+            # Always copy command index to avoid initializing it again
+            files_to_copy = ['commandIndex.json']
             # In recording mode, copy login credentials from global config dir to the dummy config dir
             if os.getenv(ENV_VAR_TEST_LIVE, '').lower() == 'true':
-                if os.path.exists(GLOBAL_CONFIG_DIR):
-                    ensure_dir(config_dir)
-                    import shutil
-                    for file in ['azureProfile.json', 'msal_token_cache.bin', 'clouds.config', 'msal_token_cache.json',
-                                 'service_principal_entries.json']:
-                        try:
-                            shutil.copy(os.path.join(GLOBAL_CONFIG_DIR, file), config_dir)
-                        except FileNotFoundError:
-                            pass
+                files_to_copy.extend([
+                    'azureProfile.json', 'clouds.config',
+                    'msal_token_cache.bin', 'msal_token_cache.json',
+                    'service_principal_entries.bin', 'service_principal_entries.json'
+                ])
+
+            ensure_dir(config_dir)
```

Comment:
`ensure_dir(config_dir)` will be called anyway later in `azure.cli.core.AzCli.__init__`:

https://github.com/Azure/azure-cli/blob/f98ba8062f08bf5b7460260c56916bd3074bf9e5/src/azure-cli-core/azure/cli/core/__init__.py#L78

We can unconditionally do it here to make the code neat.

As `config_dir` is a sub-folder in `GLOBAL_CONFIG_DIR`, `ensure_dir` will also create `GLOBAL_CONFIG_DIR`.